Readme.md

```diff
@@ -68,6 +68,18 @@ composer test-all      # psalm, compiler & core tests after each other
 > composer test-core
 ```
 
+### Architecture Decision Record (ADR) 
```

Comment:
I think this section should no be called ADR. ADR is one of the results that could happen. I would call it "Substantial changes".

Also: Move this to the contribution section.